pages/_app.js

```diff
@@ -1,10 +1,44 @@
 import { GlobalStyle } from "@/GlobalStyles";
+import { useState } from "react";
+import { pets } from "@/lib/Data";
+import { useRouter } from "next/router";
+import { uid } from "uid";
 
 export default function App({ Component, pageProps }) {
+  const [petCollection, setPetCollection] = useState(pets);
+  const router = useRouter();
+
+  function handleCreatePet(petData) {
+    setPetCollection((prevData) => [
+      {
+        id: uid(),
+        name: petData.name,
+        type: petData.type,
+        picture: petData.picture,
+        characteristics: [
+          petData.characteristic1,
+          petData.characteristic2,
+        ].filter(Boolean),
```

Comment:
I'm not sure but I think you don't need that.

Reply:
this is currently necessary as the second characteristic is optional and should only be included in the object if its value is not an empty string